docs/zkasm/infra.md

```diff
@@ -0,0 +1,64 @@
+# General Overview
+
+Firstly, we aim to implement the following types of tests and benchmarks:
+- Wasm spectest: Programs generated from Wasm spectest.
+- Benchmarks: Programs we manually write in Wasm or Rust and compile to Wasm.
+
+Both types of programs will undergo the same pipeline:
+- Compile to zkasm.
+- Run the generated zkasm & measure performance.
+- Track changes.
+- Alert in GitHub PRs in case of degradation.
+
+We use the snapshot-testing pattern in many places (link).
+
+Performance data is collected from both tests and benchmarks.
+Benchmarks also serve as global tests.
+
+## Test Creation and Generation
+We use a snapshot-based pattern for generating Wasm from spectest wast files. Benchmarks are added manually.
```

Comment:
I think we should re-consider this design decision and explore what our other options are:
- [Current] Generate WAT files from WAST files and run them through WAT -> ZKASM pipeline
- Generate ZKASM files directly from WAST files and don't commit WAT files to Git
- Don't commit ZKASM files to Git and instead pass generated ZKASM code directly to JS test runner

We need to understand what are the pros and cons of each approach and see which one is the most appropriate for our goals.

Reply:
I created an issue where I suggest to discuss if we should get rid of snapshot testing: #188